UnityProject/Assets/Scripts/HealthV2/Living/Metabolism/DigestiveSystemBase.cs

```diff
@@ -0,0 +1,198 @@
+using System.Collections.Generic;
+using System;
+using Chemistry;
+using UnityEngine;
+using Player.Movement;
+using Items.Implants.Organs;
+
+namespace HealthV2
+{
+	public class DigestiveSystemBase : MonoBehaviour, IMovementEffect
+	{
+		//Assuming 20 bodyparts (including organs) on average with a consumption of one.
+		//That is 2 Hunger per second. We want around 0.055 per second to have a full stomach (100u) last half an hour
+		//2 / 0.055 is 36.
+		private const float HUNGER_DEPLETION_DIVIDER = 36f;
+		private float hungerConsumptionRate = 20;
+
+		private bool isActive = true;
+		private float secondsToProcess = 10f;
+		public float CurrentHunger { get; private set; } = 0;
+		private float maxHunger = 100;
+
+		private List<IStomachProcess> stomachList = new List<IStomachProcess>();
+		public List<BodyFat> BodyFat { get; private set; } = new List<BodyFat>();
+
+		private LivingHealthMasterBase livingHealthMaster;
+
+		public HungerState HungerState
+		{
+			get
+			{
+				return GetCurrentHungerState();
+			}
+		}
+
+		#region MovementFields
+
+		[SerializeField] private float maxRunSpeedDebuff = -2;
+		[SerializeField] private float maxWalkingDebuff = -1.5f;
+		[SerializeField] private float maxCrawlDebuff = -0.2f;
+
+		public float RunningSpeedModifier { get; private set; }
+
+		public float WalkingSpeedModifier { get; private set; }
+
+		public float CrawlingSpeedModifier { get; private set; }
+
+		#endregion
+
+		public void OnEnable()
+		{
+			if (isActive) UpdateManager.Add(ProcessStomachs, secondsToProcess);
+		}
+
+		public void OnDisable()
+		{
+			if (isActive) UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, ProcessStomachs);
+		}
+
+		private void ProcessStomachs()
+		{
+			hungerConsumptionRate = 0;
+			foreach (BodyPart part in livingHealthMaster.BodyPartList) //Consume food from body parts
+			{
+				hungerConsumptionRate += part.HungerConsumption / HUNGER_DEPLETION_DIVIDER;
+			}
+			CurrentHunger = Math.Max(0, CurrentHunger - hungerConsumptionRate); 
```

Comment:
Body parts dont have their own hunger

Reply:
If they want to have an effect that changes with hunger, they just go DigestiveSystem.HungerState

Reply:
the modifier is just how much they drain from the hunger pool

Reply:
but the modifier is still there, and also is useful stuff like slowness when hungry and Blurry vision, 
you just need to go through and set the hunger on each body part that needs it

Reply:
in this system all you do is access the hungerstate

Reply:
hunger modifier should apply to all body parts that digest food, if there is a generic way of doing it that's good,
the hunger modifier should be on the digestive system, But the effects should apply to all the body parts that use Food

Reply:
When you say hunger modifier, are you just refering to the HungerState, if so that is inteded to be fully cenetered on digestive situation, and if a body part wants to have a hunger effect, like you want eyes to be blurry, then the eyes just grap the hunger stae from the digestive system. I cant remember if Ive already set it up like that or not, If I have not then I gotta get round to doing so as thats how its supposed to be